enterprise_access/apps/content_assignments/content_metadata_api.py

```diff
@@ -17,22 +17,49 @@
 DEFAULT_STRFTIME_PATTERN = '%b %d, %Y'
 
 
+def _process_course_metadata(course_metadata, is_run_based_by_content_key, metadata_by_key):
+    """
+    Helper function to update the metadata_by_key dictionary with the course metadata based on
+    either a course run key (for run-based assignments) or a course key.
+    """
+    course_key = course_metadata.get('key')
+
+    # If a course run is assigned, update the metadata for the course run key. Otherwise,
+    # update the metadata for the course key.
+    for assignment_content_key, is_assigned_course_run in is_run_based_by_content_key.items():
+        if is_assigned_course_run:
+            metadata_by_key[assignment_content_key] = course_metadata
+        else:
+            # if not is_assigned_course_run, we can assume it's a (legacy) course-based assignment
+            metadata_by_key[course_key] = course_metadata
```

Comment:
Aren't these still the same logic in each block? If the assignment is a legacy assignment, then `course_key` is equivalent to `assignment_content_key`, is it not?
```suggestion
        metadata_by_key[assignment_content_key] = course_metadata
```

Reply:
@pwnage101 Ahhh, I understand now; you're absolutely correct. I pushed another commit https://github.com/openedx/enterprise-access/pull/561/commits/91ac3ab7e4032481622e1808ec7f77b64018ce7d with a refactor. Let me know what you think!

Reply:
[inform] @pwnage101 I updated once more in https://github.com/openedx/enterprise-access/pull/561/commits/cb9d27bf0a90abbdabb0a3f21bfcb6b87475f3bd.